File: CSharp/Solutions/4.2. Complex Conditions - Exam Problems/MatchTickets.cs
namespace MatchTickets
{
    using System;

    public static class MatchTickets
    {
        public static void Main()
        {
            // Read input data
            decimal budget = decimal.Parse(Console.ReadLine());
            string ticketType = Console.ReadLine();
            int people = int.Parse(Console.ReadLine());

            // Initialization of the required parameters
            decimal transportCharges = 0.00M;
            decimal moneyForTickets = 0.00M;
            decimal moneyDifference = 0.00M;

            // Calculate transport charges
            if (people <= 4)
            {
                transportCharges = 0.75M * budget;
            }
            else if (people <= 9)
            {
                transportCharges = 0.60M * budget;
            }
            else if (people <= 24)
            {
                transportCharges = 0.50M * budget;
            }
            else if (people <= 49)
            {
                transportCharges = 0.40M * budget;
            }
            else if (people >= 50)
            {
                transportCharges = 0.25M * budget;
            }

            // Calculate money for tickets.
            switch (ticketType)
            {
                case "Normal":
                    moneyForTickets = people * 249.99M;
                    break;
                case "VIP":
                    moneyForTickets = people * 499.99M;
                    break;
                default:
                    moneyForTickets = people * 249.99M;
                    break;
            }

            // Calculate amount of money left and result to display.
            moneyDifference = budget - transportCharges - moneyForTickets;
            string result = string.Format(
                "Yes! You have {0:F2} leva left.",
                decimal.Round(moneyDifference, 2));

            if (moneyDifference < 0)
            {
                result = string.Format(
                    "Not enough money! You need {0:F2} leva.",
                    Math.Abs(decimal.Round(moneyDifference, 2)));
            }

            // Display the result
            Console.WriteLine(result);
        }
    }
}

File: Java/Solutions/4.2. Complex Conditions - Exam Problems/MatchTickets.java
import java.lang.*;
import java.math.BigDecimal;
import java.text.DecimalFormat;
import java.util.Scanner;

public class MatchTickets {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read input data
        BigDecimal budget = new BigDecimal(scanner.nextLine());
        String ticketType = scanner.nextLine();
        int people = Integer.parseInt(scanner.nextLine());

        // Initialization of the required parameters
        BigDecimal transportCharges = new BigDecimal("0.00");
        BigDecimal moneyForTickets = new BigDecimal("0.00");
        BigDecimal moneyDifference = new BigDecimal("0.00");

        // Calculate transport charges
        if (people <= 4) {
            transportCharges = new BigDecimal("0.75").multiply(budget);
        } else if (people <= 9) {
            transportCharges = new BigDecimal("0.60").multiply(budget);
        } else if (people <= 24) {
            transportCharges = new BigDecimal("0.50").multiply(budget);
        } else if (people <= 49) {
            transportCharges = new BigDecimal("0.40").multiply(budget);
        } else if (people >= 50) {
            transportCharges = new BigDecimal("0.25").multiply(budget);
        }

        // Calculate money for tickets.
        switch (ticketType) {
            case "Normal":
                moneyForTickets = new BigDecimal(people).multiply(new BigDecimal("249.99"));
                break;
            case "VIP":
                moneyForTickets = new BigDecimal(people).multiply(new BigDecimal("499.99"));
                break;
            default:
                moneyForTickets = new BigDecimal(people).multiply(new BigDecimal("249.99"));
                break;
        }

        // Calculate amount of money left and result to display.
        DecimalFormat decimalFormatter = new DecimalFormat("0.00");
        moneyDifference = budget.subtract(transportCharges.add(moneyForTickets));
        String result = String.format("Yes! You have %s leva left.", decimalFormatter.format(moneyDifference.setScale(2, BigDecimal.ROUND_HALF_UP)));

        if (moneyDifference.compareTo(BigDecimal.valueOf(0)) < 0) {
            result = String.format("Not enough money! You need %s leva.", decimalFormatter.format(moneyDifference.setScale(2, BigDecimal.ROUND_HALF_UP).abs()));
        }

        // Display the result
        System.out.println(result);
    }
}
